Source/OrchBase/OrchBaseLink.cpp

```diff
@@ -144,7 +144,178 @@ if (pT->linked) {                      // Task already linked?
 
 */
 // Place and set the link flag if the placement was successful.
-if (!pPlace->Place(pT)) pT->LinkFlag();
+// if (!pPlace->Place(pT)) pT->LinkFlag();
+if (!pPlace->Place(pT))
+{
+   // send data to NameServer. Once the P_builder is threaded, this
+   // should be moved to the build command.
+   int nsRank = Q::NAP;
+   MPI_Comm nsComm = MPI_COMM_NULL;
+   vector<pair<unsigned, int> >superRanks;
+   for (unsigned comm = 0; comm < Comms.size(); comm++)
+   {
+       WALKVECTOR(int, pPmap[comm]->U.Mothership, m)
+          superRanks.push_back(pair<unsigned,int>(comm, *m));
+       if ((nsRank = pPmap[comm]->U.NameServer) != Q::NAP)
+       {
+          nsComm = Comms[comm];
+          break;
+       }
+   }
+   if (nsRank == Q::NAP)
+   {
+      Post(711); // No nameserver. This should be fatal 
+      return;    // (need to return with a value from TaskLoad)
+   }
+   if (nsComm == MPI_COMM_NULL)
+   {
+      Post(712); // Unknown comm for nameserver. VERY fatal
+      return;    // (should abort)
+   }
+   PMsg_p deviceData(nsComm);
+   vector<string> supervisorNames;
+   vector<SymAddr_t> supervisorAddrs;
+   vector<unsigned long> supervisorRanks;
+   string supervisorType = pT->pSup->pP_devtyp->Name();
+   vector<AttrIdx> supervisorAttrs;
+   // vector<RecordData_t> supervisorDevices;
+   vector<pair<unsigned, int> >::iterator currSuperRank = superRanks.begin();
+   deviceData.Src(Urank);
+   deviceData.Tgt(nsRank);
+   deviceData.Zname(0,pT->Name());
+   WALKMAP(AddressComponent,P_box*,pE->P_boxm,box)
+   {
+      // find and set up supervisor devices 
+      vector<P_super*>::iterator superIter = box->second->P_superv.begin();
+      while (((*superIter) != pT->pSup) && (++superIter != box->second->P_superv.end()));
+      if (superIter == box->second->P_superv.end())
+      {
+	 Post(164,pT->Name());
+	 return;

```

Comment:
Good catch. I thought initially that simply by not setting the linkflag, you could prevent the system from 'seeing' the task as linked, but you would have to clear out the thread-device lists as well, which are already being set, because placement simply appends to the list, it doesn't overwrite it and start again (is this the correct behaviour? I'm not going to voice an opinion on that point) when you invoke Place for the task.